inference-engine/tests/unit/vpu/middleend_tests/passes_tests/adjust_data_batch_tests.cpp

```diff
@@ -0,0 +1,389 @@
+        pipeline.addPass(passManager->dumpModel("adjustDataBatch"));
+        pipeline.run(testModel.getBaseModel());
+    }
+
+    DataVector checkSingleLoopStart(const Data& data) {
+        EXPECT_EQ(data->desc().dim(Dim::N), batchSize);
+        EXPECT_EQ(data->numConsumers(), 2);
+
+        DataVector outputs;
+        for (const auto& consumer : data->consumers()) {
+            EXPECT_TRUE(consumer->type() == StageType::LoopStart || consumer->type() == StageType::LoopEnd);
+            if (consumer->type() == StageType::LoopStart) {
+                for (const auto& output : consumer->outputs()) {
+                    EXPECT_EQ(output->desc().dim(Dim::N), 1);
+                    outputs.push_back(output);
+                }
+            }
+        }
+
+        return outputs;
+    }
+
+    DataVector checkBranches(const Data& root, const std::vector<StageType>& consumersTypes) {
+        auto successors = DataVector{};
+
+        const auto& consumers = root->consumers() | asVector();
+        EXPECT_EQ(consumers.size(), consumersTypes.size());
+        for (std::size_t i = 0; i < consumers.size(); ++i) {
+            const auto& consumer = consumers[i];
+            const auto& expected = consumersTypes[i];
+            EXPECT_EQ(consumer->type(), expected);
+
+            EXPECT_EQ(consumer->numOutputs(), 1);
+            const auto& output = consumer->output(0);
+            successors.push_back(output);
+
+            if (expected == StageType::LoopStart) {
+                EXPECT_EQ(consumer->numOutputs(), 1);
+                EXPECT_EQ(output->desc().dim(Dim::N), 1);
+            } else if (expected == StageType::LoopEnd) {
+                EXPECT_EQ(output->desc().dim(Dim::N), batchSize);
+            }
+        }
+
+        return successors;
+    }
+
+    DataVector checkSingleLoopEnd(const Data& data) {
+        EXPECT_EQ(data->numConsumers(), 1);
+
+        const auto& consumer = data->singleConsumer();
+        EXPECT_EQ(consumer->type(), StageType::LoopEnd);
+        DataVector outputs;
+        for (const auto& output : consumer->outputs()) {
+            EXPECT_EQ(output->desc().dim(Dim::N), batchSize);
+            outputs.push_back(output);
+        }
+
+        return outputs;
+    }
+
+    static Data CheckSingleConnection(const Data& data, int testInd, int batch = 1) {
+        EXPECT_EQ(data->numConsumers(), 1);
+
+        const auto& consumer = data->singleConsumer();
+        EXPECT_EQ(consumer->type(), StageType::None);
+        EXPECT_EQ(consumer->attrs().get<int>("test_ind"), testInd);
+        EXPECT_EQ(consumer->numOutputs(), 1);
+        const auto& output = consumer->output(0);
+        EXPECT_EQ(output->desc().dim(Dim::N), batch);
+        return output;
+    }
+
+    static Data singleElement(const DataVector& dataObjects) {
+        EXPECT_EQ(dataObjects.size(), 1);
+        return dataObjects.front();
+    }
+};
+
+TEST_F(VPU_AdjustDataBatchTest, Case1_LinearWithBatchedInTheEnd) {
```

Comment:
Nitpick: I don't like `Case#` in the tests' name, if the cases are different, just reflect it in its name.